src/data_info/services/stat.clj

```diff
@@ -1,201 +1,136 @@
-    :read  (validators/all-paths-readable cm user paths)
-    (validators/all-paths-readable cm user paths)))
+    :own   #{:own}
+    :write #{:own :write}
+    #{:own :write :read}))
 
-(defn remove-inaccessible-paths
+(defn- remove-inaccessible-paths
   "Removes entries from a list of paths that the user cannot access."
-  [cm user validation-behavior paths]
-  (case (keyword validation-behavior)
-    :own   (filter (partial perm/owns? cm user) paths)
-    :write (filter (partial perm/is-writeable? cm user) paths)
-    :read  (filter (partial perm/is-readable? cm user) paths)
-    (filter (partial perm/is-readable? cm user) paths)))
+  [irods user validation-behavior paths]
+  (let [acceptable-permissions (acceptable-permissions-for validation-behavior)]
+    (filter
+     (fn [p] (get acceptable-permissions @(rods/permission irods user (cfg/irods-zone) p)))
+     paths)))
 
 (defn do-stat
   [{:keys [user validation-behavior filter-include filter-exclude ignore-missing ignore-inaccessible]
     :or   {filter-include nil filter-exclude nil ignore-missing false ignore-inaccessible false}}
    {paths :paths uuids :ids}]
-  (irods/with-jargon-exceptions [cm]
-    (validators/user-exists cm user)
-    (when-not ignore-missing (validators/all-uuids-exist cm uuids))
-    (let [uuid-paths       (get-uuid-paths cm uuids)
-          all-paths        (into paths (map second uuid-paths))
-          extant-paths     (if ignore-missing
-                             (remove-missing-paths cm all-paths)
-                             (do (validators/all-paths-exist cm all-paths) all-paths))
+  (irods/with-irods-exceptions {} irods
+    (validate irods
+              [:user-exists user (cfg/irods-zone)]
+              (when-not ignore-missing [:uuid-exists (remove nil? uuids)])
+              (when-not ignore-missing [:path-exists (remove nil? paths) (cfg/irods-user) (cfg/irods-zone)]))
+    (let [uuid-paths       @(rods/uuids->paths irods uuids)
+          all-paths        (into paths (remove nil? (map second uuid-paths)))
+          extant-paths     (remove-missing-paths irods user (cfg/irods-zone) all-paths)
           accessible-paths (if ignore-inaccessible
-                             (remove-inaccessible-paths cm user validation-behavior extant-paths)
-                             (do (check-stat-permissions cm user validation-behavior extant-paths) extant-paths))
+                             (remove-inaccessible-paths irods user validation-behavior extant-paths)
+                             (do (check-stat-permissions irods user validation-behavior extant-paths) extant-paths))
           uuid-paths       (filter (comp (set accessible-paths) second) uuid-paths)
           paths            (filter (set accessible-paths) paths)
           format-stat      (fn [path]
-                             (path-stat cm user path :filter-include filter-include :filter-exclude filter-exclude))]
+                             ;; TODO: update this to use clj-irods
```

Comment:
Am I missing something, or is this using clj-irods now/the comment can be removed?

Reply:
Ah, that can be removed. Thanks for catching that.